codex-cli/bin/codex.js

```diff
@@ -65,9 +65,40 @@ const binaryPath = path.join(__dirname, "..", "bin", `codex-${targetTriple}`);
 // receives a fatal signal, both processes exit in a predictable manner.
 const { spawn } = await import("child_process");
 
+async function tryImport(moduleName) {
+  try {
+    // eslint-disable-next-line node/no-unsupported-features/es-syntax
+    return await import(moduleName);
+  } catch (err) {
+    console.error(`unable to import ${moduleName}`, err);
+    return null;
+  }
+}
+
+async function resolveRgPath() {
+  const { rgPath } = await tryImport("@vscode/ripgrep");
+  if (!rgPath) {
+    throw new Error("ripgrep not found");
+  }
+  return path.dirname(rgPath);
+}
+
+function getUpdatedPath(newDirs) {
+  const pathSep = process.platform === "win32" ? ";" : ":";
+  const existingPath = process.env.PATH || "";
+  const updatedPath = [
+    ...newDirs,
+    ...existingPath.split(pathSep).filter(Boolean),
+  ].join(pathSep);
+  return updatedPath;
+}
+
+const rgPath = await resolveRgPath();
+const updatedPath = getUpdatedPath([rgPath]);
```

Comment:
Should we throw in this case? I feel like, if the user has installed with `--ignore-scripts` or the network request failed to fetch `rg` for some reason, we should still let them run the CLI?